lighthouse-core/test/fraggle-rock/user-flow-test.js

```diff
@@ -0,0 +1,126 @@
+      'Navigation report (example.com/)',
+      'Search input',
+      'Search results',
+      'Navigation report (example.com/page)',
+    ]);
+  });
+
+  it('enumerates if multiple in same group with no name', () => {
+    const steps = /** @type {any} */ ([
+      {lhr: {gatherMode: 'navigation', finalUrl: 'https://example.com'}},
+      {lhr: {gatherMode: 'timespan', finalUrl: 'https://example.com'}},
+      {lhr: {gatherMode: 'snapshot', finalUrl: 'https://example.com'}},
+      {lhr: {gatherMode: 'snapshot', finalUrl: 'https://example.com'}},
+    ]);
+
+    // @ts-expect-error page will not be use, we are only testing _getDerivedStepNames
+    const flow = new UserFlow();
+    flow.steps = steps;
+
+    const stepNames = flow._getDerivedStepNames();
+
+    expect(stepNames).toEqual([
+      'Navigation report (example.com/)',
+      'Timespan report (example.com/)',
+      'Snapshot report 1 (example.com/)',
+      'Snapshot report 2 (example.com/)',
+    ]);
+  });
+
+  it('resets count in new group', () => {
+    const steps = /** @type {any} */ ([
+      {lhr: {gatherMode: 'navigation', finalUrl: 'https://example.com'}},
+      {lhr: {gatherMode: 'timespan', finalUrl: 'https://example.com'}},
+      {lhr: {gatherMode: 'snapshot', finalUrl: 'https://example.com'}},
+      {lhr: {gatherMode: 'snapshot', finalUrl: 'https://example.com'}},
+      {lhr: {gatherMode: 'navigation', finalUrl: 'https://example.com/page'}},
+      {lhr: {gatherMode: 'snapshot', finalUrl: 'https://example.com/page'}},
+    ]);
+
+    // @ts-expect-error page will not be use, we are only testing _getDerivedStepNames
+    const flow = new UserFlow();
+    flow.steps = steps;
+
+    const stepNames = flow._getDerivedStepNames();
+
+    expect(stepNames).toEqual([
+      'Navigation report (example.com/)',
+      'Timespan report (example.com/)',
+      'Snapshot report 1 (example.com/)',
+      'Snapshot report 2 (example.com/)',
+      'Navigation report (example.com/page)',
+      'Snapshot report (example.com/page)',
+    ]);
+  });
+
+  it('does not increment if name is provided', () => {
+    const steps = /** @type {any} */ ([
+      {lhr: {gatherMode: 'navigation', finalUrl: 'https://example.com'}},
+      {lhr: {gatherMode: 'timespan', finalUrl: 'https://example.com'}},
+      {lhr: {gatherMode: 'snapshot', finalUrl: 'https://example.com'}},
+      {lhr: {gatherMode: 'snapshot', finalUrl: 'https://example.com'}, name: 'Search results'},
+      {lhr: {gatherMode: 'snapshot', finalUrl: 'https://example.com'}},
+    ]);
+
+    // @ts-expect-error page will not be use, we are only testing _getDerivedStepNames
+    const flow = new UserFlow();
+    flow.steps = steps;
+
+    const stepNames = flow._getDerivedStepNames();
+
+    expect(stepNames).toEqual([
+      'Navigation report (example.com/)',
+      'Timespan report (example.com/)',
+      'Snapshot report 1 (example.com/)',
+      'Search results',
+      'Snapshot report 2 (example.com/)',
```

Comment:
I'm concerned the rules for making these names are complex, and the outcome may be confusing. ex: here, there are 3 snapshots but the third one is called "Snapshot report 2". Throwing out some other ideas...

* If no user-provided name, prefix the step with `${stepIndex+1}: `–ex: `#5: Snapshot report  (example.com/)`\* . Or just always prefix it.
* Just leave as `Snapshot report (example.com/)` and rely on the UI to specify which number step it is to distinguish

\* Just realized that this method, and the current one in the PR, are both lacking in terms of i18n.

Reply:
How do you feel about using global counters for timespan and snapshot, getting rid of the navigation groups. So we would have:

```
Navigation report (example.com/)
Timespan report 1 (example.com/)
Snapshot report 1 (example.com/)
Search results // Name snapshot
Snapshot report 3 (example.com/)

Navigation report (example.com/page)
Timespan report 2 (example.com/page)
Snapshot report 4 (example.com/page)
```

Reply:
> Just realized that this method, and the current one in the PR, are both lacking in terms of i18n.

Does putting the index in front help i18n?

Reply:
> How do you feel about using global counters for timespan and snapshot

I think this is better than the current impl, yea!

> Does putting the index in front help i18n?

Stuffing ordinal information into these default strings means we can't i18n it. I think we need to discuss putting this information in the flow UI somehow, so it can actually be localized. These names don't really need to be unique/distinguishable _in the data_, just in the view.

Reply:
> How do you feel about using global counters for timespan and snapshot
I think this is better than the current impl, yea!

I think this discussion is focused a bit too much on how this makes sense on the test/script side and missing how these names are being presented to report consumers, which is very much in the context of each page/navigation.

The UX study specifically found that users view these reports segmented by page (so much so in fact that they mistakenly believe the navigation report is summarizing the nested reports as well). From that lens, seeing the first and only snapshot for a particular page being labelled "Snapshot 6" does not make a lot of sense and muddles the segmented mental model we should be leaning into.

I'd strongly prefer keeping the per-page logic without new evidence that our users prefer the flow-wide interpretation proposed here.

Reply:
Assuming we stick with the per-page logic, we still need to figure out how to handle i18n. I've added i18n as an action item in #11313.

> Stuffing ordinal information into these default strings means we can't i18n it. I think we need to discuss putting this information in the flow UI somehow, so it can actually be localized.

Do we have a way to i18n ordinals on their own?

Reply:
> first and only snapshot for a particular page being labelled "Snapshot 6"

it would be Snapshot 1, from my understanding. and if followed by a named snapshot, the next unnamed snapshot would be snapshot 3 (as opposed to Snapshot 2 in the original impl). "Snapshot 6" for the first snapshot but 6th step is not good / what I was thinking of.

Reply:
> in other words by "global counter" i believe adam meant not resetting the counts for every navigation.

Yes, I think we have the same understanding here. In this case, if we have 

```
Navigation report (example.com/)
Snapshot report 1 (example.com/)
Search results // Name snapshot
Snapshot report 3 (example.com/)

Navigation report (example.com/product)
Snapshot report 4 (example.com/product)
Snapshot report 5 (example.com/product)

Navigation report (example.com/checkout)
Snapshot report 6 (example.com/checkout)
```

My comment was that the first and only snapshot for `/checkout` is labelled "Snapshot report 6" in this case, which is undesirable given what we know about how users understand the reports.

Removing numbers sounds great too, I'd rather have no number at all than global numbering.

Reply:
I'm thinking we remove the enumeration then. I'll add a note for the UX sync to discuss an i18n friendly way to identify unnamed reports.